packages/commonwealth/server.ts

```diff
@@ -23,16 +23,16 @@ import { DatabaseCleaner } from './server/util/databaseCleaner';
 import 'express-async-errors';
 
 // bootstrap production adapters
+if (config.APP_ENV === 'production') {
+  stats(HotShotsStats());
+  analytics(MixpanelAnalytics());
+  if (config.NOTIFICATIONS.FLAG_KNOCK_INTEGRATION_ENABLED)
+    notificationsProvider(KnockProvider());
+}
```

Comment:
I generally think that environment-dependent settings should be set via env vars so that the code doesn't ever need to know which environment it's in. e.g. `STATS_ADAPTER` and `ANALYTICS_ADAPTER` in this case.

My main reasoning for this is that we have multiple places in the codebase where we have this kind of logic, but env vars are centralized.

Reply:
That's one way to reason about configuration. But, I don't think having these "localized" config sections is a big issue, if we can avoid the proliferation of .env variables for every config decision we have to make.  I personally prefer to encapsulate most of these in the "top" application layer though, very close to the "main" entry point.

Reply:
I actually think @rbennettcw might be correct here. Per my other comment, filtering by `APP_ENV === 'production'` is not a good idea. Rather if we have `STATS_ADAPTER` and `ANALYTICS_ADAPTER` we can refine the Zod schema so that when `APP_ENV === 'production'` then those 2 env var must be set to a non-default value. This ensures adapters are correctly set in production without preventing the usage of those adapters in non-production environments.

Reply:
Maybe I'm missing a scenario where we want to collect analytics and stats in a non-prod environment. Can you describe when this is necessary and why?

Reply:
Using the local StatsD server you can simulate the actual behaviour of stats reporting which is useful for debugging stats e.g. failure to report (process exit before request completes), incorrect stats/format, Datadog debugging, etc.